### PR TITLE
Don't always Enable SSHD and Open SSH Port [SLE-15-SP4]

### PR DESCRIPTION
**Pull request by shundhammer**
# This was superseded by #1090.

### CODE REVIEW

File: src/lib/installation/security_settings.rb [hunk 1]
```diff
@@ -67,6 +67,30 @@ def load_features
       load_feature(:polkit_default_privs, :polkit_default_privileges)
     end
 
+    # Return whether the sshd service will be enabled.
+    #
+    # @return [Boolean]
+    def enable_sshd
```
Comment (teclator) [inline src/lib/installation/security_settings.rb:73]: To be honest I don't like this approach and it is a getter that modify the state of the object which is not expected . For enabling the sshd there is already a bang method. I would maintain the getters as them are.

File: src/lib/installation/security_settings.rb [hunk 2]
```diff
@@ -181,27 +205,32 @@ def global_section
     end
 
     def wanted_enable_sshd?
-      Yast::Linuxrc.usessh || only_public_key_auth || @enable_sshd
```
Comment (teclator) [inline src/lib/installation/security_settings.rb:184]: This is the main fix, we should not evaluate if we only have public key access during the initialization but when the proposal is done, therefore, we could do it as part of the proposal and for that we probably will need to extend the state to know if it was already proposed or not.

**Comment by teclator** [comment]
Until now, the **SecuritySettings** instance was created at the proposal summary step or by **AutoYaST** when it was needed and after the users configuration was already done. But since we [added](https://github.com/yast/yast-security/pull/128) security policies support (STIG) this is different and it is evaluated too early mainly because it also create and instance of **SecuritySettings** during the initialization instead of doing it using a lazy load init.

We already have a **propose_lsm_config** as part of the initialization, maybe we should move that proposal and the ssh config proposal to a **propose method** which will be called when it is really needed storing that the propose is already done just to avoid resetting it.

**Comment by teclator** [comment]
The other options is fix the initialization of the SecurityPolicies::TargetConfig moving the @security accessor to a method 

 https://github.com/yast/yast-security/pull/128/files#diff-b6b853b94bce3cd38105d93a59732aa0345ed6ad7e53df5afad8e337d13bdb7cR54

File: src/lib/installation/security_settings.rb (hunk 2 above)
Comment (shundhammer) [reply src/lib/installation/security_settings.rb:184]: Then we'd have to rely on somebody from the outside to do the right thing. That is a lot less reliable.

**Pull request closed without merging**